src/http-api/errors.ts

```diff
@@ -63,11 +67,66 @@ export class MatrixError extends HTTPError {
         if (url) {
             message = `${message} (${url})`;
         }
-        super(`MatrixError: ${message}`, httpStatus);
+        super(`MatrixError: ${message}`, httpStatus, httpHeaders);
         this.errcode = errorJson.errcode;
         this.name = errorJson.errcode || "Unknown error code";
         this.data = errorJson;
     }
+
+    /**
+     * @returns whether this error is due to rate-limiting.
+     */
+    public isRateLimitError(): boolean {
+        return this.errcode === "M_LIMIT_EXCEEDED" || (this.errcode === "M_UNKNOWN" && this.httpStatus === 429);
+    }
+
+    /**
+     * @returns the recommended delay in milliseconds to wait before retrying
+     * the request that triggered this error, or null if no delay is recommended.
+     * @throws Error if the recommended delay is an invalid value.
+     */
```

Comment:
Maybe add a link to `safeGetRetryAfterMs` here, to help people find it

Reply:
Done in e565d5da3